### ISSUE TITLE
WebSocket messages do not occur with a Zone

### ISSUE THREAD

**Issue by matthewp** [canjs/can-zone#108]
```js
ws.onmessage = function(){

};
```

If some behavior is completed within a websocket message this needs to be wrapped within the Zone.

**Comment by frank-dspeed** [comment]
is this the bug https://github.com/feathersjs/feathers-socketio/issues/81

they say zones can't handle socket.io in general.

**Comment by matthewp** [comment]
Yep, this is true.

**Comment by frank-dspeed** [comment]
@matthewp thanks a lot you marked this a bug so you think this is solve able? does this work with any other object that is a eventEmitter because socket.io is nothing else.

**Comment by frank-dspeed** [comment]
@matthewp sorry secund question do you know if this issue also happens with other socket libs like primus?

**Comment by frank-dspeed** [comment]
as far as i understand 
For can-zone to work we have to override various task-creating functionality, this is the list of what we currently implement:

Macrotasks

setTimeout
XMLHttpRequest
Microtasks

requestAnimationFrame
Promise
process.nextTick

we need to implament socket.io here or i use a promis based socket io lib?

**Comment by matthewp** [comment]
I think the socket.io issue is because they do a recursive setTimeout. Without some heuristic to detect this I'm not sure that's fixable.

**Comment by frank-dspeed** [comment]
@matthewp last question about this when we ignore socket.io with the Interval ignore all should work right?

**Comment by matthewp** [comment]
Yes, using `Zone.ignore` will ignore any async tasks that are called within.

**Comment by frank-dspeed** [comment]
@matthewp https://github.com/donejs/donejs/issues/976

please make suggestion about that

**Comment by frank-dspeed** [comment]
https://github.com/donejs/autorender/pull/68
could bypass that via done-autorender
